main/src/ca/uwaterloo/flix/language/phase/unification/SetFormulaAlg.scala

```diff
@@ -26,8 +22,6 @@ class SetFormulaAlg {
     mkNot(f)
 
   def mkAnd(f1: SetFormula, f2: SetFormula)(implicit universe: Set[Int]): SetFormula = (f1, f2) match {
-    case (All, other) => other
```

Comment:
I think these should be preserve since they can remove a whole term, just check for `== universe` instead